src/main/scala/com/fulcrumgenomics/util/ReadStructure.scala

```diff
@@ -139,10 +139,14 @@ class ReadStructure(segs: Seq[ReadSegment], resetOffsets: Boolean = false) exten
   }
 
   private def collectSegments[T <: ReadSegment: ClassTag]: Seq[T] = this.collect { case t: T => t }
-  def template: Seq[Template] = this.collectSegments[Template]
-  def sampleBarcode: Seq[SampleBarcode] = this.collectSegments[SampleBarcode]
+  def template: Seq[Template]                 = this.collectSegments[Template]
+  def sampleBarcode: Seq[SampleBarcode]       = this.collectSegments[SampleBarcode]
   def molecularBarcode: Seq[MolecularBarcode] = this.collectSegments[MolecularBarcode]
-  def skip: Seq[Skip] = this.collectSegments[Skip]
+  def skip: Seq[Skip]                         = this.collectSegments[Skip]
+  def templateStructure(resetOffsets: Boolean = false): ReadStructure         = new ReadStructure(this.template, resetOffsets=resetOffsets)
```

Comment:
I didn't see any usages of this with `reset=true`, so I'm not sure it's needed vs. just the `def template`.  Same for the others below.